offload/libomptarget/omptarget.cpp

```diff
@@ -798,7 +798,15 @@ int processAttachEntries(DeviceTy &Device, AttachInfoTy &AttachInfo,
                       << ", PtrSize=" << PtrSize << ", MapType=0x"
                       << llvm::utohexstr(MapType);
 
-    const bool IsAttachAlways = MapType & OMP_TGT_MAPTYPE_ALWAYS;
+    bool IsAttachAlways = MapType & OMP_TGT_MAPTYPE_ALWAYS;
+
+    // Treat ATTACH(auto) as ATTACH(always) if environment variable is set
+    if (!IsAttachAlways && MappingConfig::get().TreatAttachAutoAsAlways) {
+      IsAttachAlways = true;
+      ODBG(ODT_Mapping) << "ATTACH(auto) will be treated as ATTACH(always) "
```

Comment:
Won't this happen at every single attach? Isn't that too verbose? Maybe it's better to do just print it once when TreatAttachAutoAsAlways is set to true?